core/src/main/scala/net/ruippeixotog/scalascraper/browser/JsoupBrowser.scala

```diff
@@ -42,25 +47,25 @@ class JsoupBrowser(val userAgent: String = "jsoup/1.8", val proxy: JavaProxy = n
     executePipeline(Jsoup.connect(url).method(POST).proxy(proxy).data(form.asJava))
 
   def parseFile(file: File, charset: String): JsoupDocument =
-    JsoupDocument(Jsoup.parse(file, charset))
+    JsoupDocument(Jsoup.parse(file, charset, baseUri, parser))
 
   def parseString(html: String): JsoupDocument =
-    JsoupDocument(Jsoup.parse(html))
+    JsoupDocument(Jsoup.parse(html, baseUri, parser))
 
   def parseInputStream(inputStream: InputStream, charset: String): JsoupDocument =
-    using(inputStream) { _ => JsoupDocument(Jsoup.parse(inputStream, charset, "")) }
+    using(inputStream) { _ => JsoupDocument(Jsoup.parse(inputStream, charset, baseUri, parser)) }
 
-  def cookies(url: String) = cookieMap.toMap
+  def cookies(url: String): Map[String, String] = cookieMap.toMap
 
-  def setCookie(url: String, key: String, value: String) = {
+  def setCookie(url: String, key: String, value: String): mutable.Map[String, String] = {
```

Comment:
We should definitely return `Unit` here.

Reply:
done